### PR TITLE
Implement "child text content" concept; use it where appropriate.

### PR DESCRIPTION
**Pull request by frewsxcv**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14549)

### CODE REVIEW

**Comment by Ms2ger** [comment]
Needs a test for the time case at least

**Comment by frewsxcv** [comment]
> Needs a test for the time case at least

Done.

File: components/script/dom/node.rs [hunk 1]
```diff
@@ -1788,6 +1788,11 @@ impl Node {
         copy
     }
 
+    /// https://html.spec.whatwg.org/multipage/#child-text-content
+    pub fn child_text_content(&self) -> DOMString {
+        Node::collect_text_contents(self.children())
```
Comment (nox) [inline components/script/dom/node.rs]: Can't you just remove `Node::collect_text_contents` altogether?

Comment (frewsxcv) [inline components/script/dom/node.rs]: It's used [here](https://github.com/servo/servo/blob/87f7b29d65d8a0ad137e38412aef50734b24ea6c/components/script/dom/node.rs#L1952).

File: components/script/dom/htmltimeelement.rs [hunk 2]
```diff
@@ -44,10 +43,7 @@ impl HTMLTimeElementMethods for HTMLTimeElement {
         if element.has_attribute(&local_name!("datetime")) {
             return element.get_string_attribute(&local_name!("datetime"))
         } else {
-            match element.GetInnerHTML() {
-                Ok(x) => x,
-                _ => DOMString::new(),
-            }
+            element.upcast::<Node>().child_text_content()
```
Comment (Ms2ger) [inline components/script/dom/htmltimeelement.rs]: Well isn't this funny. Let's see what the spec says:

> The dateTime IDL attribute must reflect the element's datetime content attribute.

Nothing in there about the text content. There is something about it in <https://html.spec.whatwg.org/multipage/semantics.html#dom-time-datetime>, but that's something else entirely, and I'm not sure it's exposed to JS.

Let's revert everything related to the time element here, and fix this method separately.

Comment (frewsxcv) [inline components/script/dom/htmltimeelement.rs]: Oh, I was reading the wrong section. I was looking at https://html.spec.whatwg.org/multipage/semantics.html#datetime-value.

Comment (frewsxcv) [inline components/script/dom/htmltimeelement.rs]: `<time>` things have been removed.

**Pull request merged**